e2e/specs/formsDashboard.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { test } from "../core";
+import { expect } from "@playwright/test";
+import { FormBuilderPage } from "../pages";
+import {
+  createForm,
+  createValueReference,
+  addFormResources,
+  deleteForm,
+} from "../commands/formOperations";
+import { Form } from "../../src/types";
+
+let form: Form = null;
+test.beforeEach(async ({ api }) => {
+  form = await createForm(api);
+  const valueReference = await createValueReference(api);
+  await addFormResources(api, valueReference, form.uuid);
+});
+
+test("Filter forms based on publish status", async ({ page }) => {
+  const formBuilderPage = new FormBuilderPage(page);
+  await test.step("When I Visit the form builder dashboard", async () => {
+    await formBuilderPage.gotoFormBuilder();
+  });
+
+  // Test the filter functionality
+  await test.step("Then I click the publish filter dropdown", async () => {
+    await page
+      .getByRole("button", { name: "Filter by publish status: All Open menu" })
+      .click();
+  });
+
+  await test.step("And I click the Unplished option", async () =>
+    await page.getByText("Unpublished").click());
+
+  // Locate the table
+  await page.$('[data-testid ="forms-table"]');
+
+  // Expect the publish status to be "No"
+  const tagElement = await await page.$(`[data-testid="no-tag"]`);
```

Comment:
```suggestion
  const tagElement = await page.getByTestId("no-tag");
```

Reply:
Using getByTestId is failing to get the innerText and they are recommending that first use case.